### ISSUE TITLE
ToJson counterpart for JsonObject.OfJson

### ISSUE THREAD

**Issue by mt-caret** [fsprojects/Fleece#70]
When serializing to and deserializing from JSON, I've often found it convenient to forgo serialization/deserialization for certain fields like so:

```fsharp
type Foo =
    { Foo : string array
    ; Bar : JsonObject
    }
```

While writing the `OfJson` member is straightforward:

```
    static member inline OfJson json =
        match json with
        | JObject o ->
            (fun f b -> { Foo = f; bar = b})
                <!> (o .@ "foo")
                <*> (o .@ "bar")
        | x -> Decode.Fail.objExpected x
```

Writing the `ToJson` member is impossible AFAIK as `JsonObject` does not have a `ToJson` static member. Is it possible to add this to Fleece?

Additionally, are there any reasons that `JsonValue` doesn't have `ToJson` and `OfJson` instances? Will a PR adding these be welcomed?

**Comment by gusty** [comment]
I wasn't that involved in the initial design, so I can't tell exactly why JsonObject doesn't have a ToJson counterpart.

The issue could be that for some implementations it's an alias (ie: FSharp.Data) while for another implementations it's a type of the underlying Json library.

For JsonValue this is also true and this issue https://github.com/mausch/Fleece/issues/55 proposes to use another type which is common to all implementations.

I think the initial design was done avoiding unnecessary wrapping with performance in mind.

However I've found a technique to create compile-time wrappers which are erased at run-time, so we can revisit these decisions with this in mind.

Anyway, if you want to work in a PR you are more than welcome and if you need help please shout as it would be interesting to know more about the problems of implementing this.

**Comment by gusty** [comment]
Closed by #71
